### DISCUSSION TITLE
Applying a RegexHighlighter with a dfeined Theme to a Log

### DISCUSSION THREAD

**Discussion by b0bh00d** [Textualize/textual#4490] [Q&A]
Perhaps I'm not seeing all the pieces for this, but I'm not seeing how I can apply a RegexHighlighter:

```
class LogHighlighter(RegexHighlighter):
    """Apply style to anything that looks like an loguru message."""

    base_style = "snlog."
    highlights = [r"(?P<date>\d+-\d+-\d+)",
                  r"(?P<time>\d+:\d+:\d+\.\d+)",
                  r"(?P<warning>WARN)",
                  r"(?P<error>ERROR)",
                  r"(?P<fatal>FATAL)",
                  r"(?P<critical>CRITICAL)",
                  r"(?P<source>\w+:[\w_]+:\d+)",
                  r"(?P<path>^/[\w/]+)",
                  ]

```

with a theme:

```
        rich.theme.Theme({
                "snlog.date": "green",
                "snlog.time": "green",
                "snlog.warning": "yellow",
                "snlog.error": "red",
                "snlog.fatal": "bold red",
                "snlog.critical": "bold magenta",
                "snlog.source": "cyan",
                "snlog.path": "bold red",
              })
```

To a Textual Log widget.  The widget has a `highlight` parameter to enable highlighting, but there are no further arguments provided to apply a different highlighter w/ a theme.  The reference page for the Log widget shows [a highlighter instance](https://textual.textualize.io/widgets/log/#textual.widgets._log.Log.highlighter) value, but I'm not sure if I am supposed to simply replace it.  I've tried doing that, but without defining the theme, the text in the Log widget is just plain white.

I humbly request instruction to wash away my ignorance, or a means added to Log to defined this more formally.

**Comment by willmcgugan** [comment]
Textual doesn't expose the `rich.theme`, so there is no straightforward way of doing that.

You can, hover, disable automatic highlighting, and log `Text` instances. Rich's `Text` objects have a number of ways of applying styles, so you can highlight however you like.

**Reply by b0bh00d** [reply]
Thanks!  I will give that a try.
